pkg/api/rest/rest.go

```diff
@@ -150,6 +151,16 @@ type Redirector interface {
 	ResourceLocation(ctx api.Context, id string) (remoteLocation *url.URL, transport http.RoundTripper, err error)
 }
 
+// Proxier supports proxying of connections
+type Proxier interface {
```

Comment:
> How does this help me implement "GET /pod/foo/proxy"?

The subresource implements Proxier (not Getter) https://github.com/csrwng/kubernetes/pull/16/files#diff-069492687bcccc75f3afb6292e6b19e7R196



> The sub resource "PodProxy" would implement Proxier and Get?

No, only proxier which tells it which methods to respond to


> How would the Get method be implemented?

It isn't


> If the Get method is implemented, wouldn't it take priority over proxier?

Both would be used (proxier would still work only with the redirect URL), Get can do whatever it wants



> Remember for proxying you eventually need the sub path, in which case you need to pass an alternate resource down to the call.

This is handled in the proxy handler. The code in this PR is functional.